main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -1087,20 +1086,36 @@ object Namer {
   /**
     * Names the given pattern `pat0`.
     */
-  private def visitPattern(pat0: WeededAst.Pattern)(implicit flix: Flix): NamedAst.Pattern = pat0 match {
-    case WeededAst.Pattern.Wild(loc) => NamedAst.Pattern.Wild(loc)
+  private def visitPattern(pat0: WeededAst.Pattern)(implicit flix: Flix): Validation[NamedAst.Pattern, NameError] = pat0 match {
+    case WeededAst.Pattern.Wild(loc) => NamedAst.Pattern.Wild(loc).toSuccess
     case WeededAst.Pattern.Var(ident, loc) =>
       // make a fresh variable symbol for the local variable.
       val sym = Symbol.freshVarSym(ident, BoundBy.Pattern)
-      NamedAst.Pattern.Var(sym, loc)
+      NamedAst.Pattern.Var(sym, loc).toSuccess
 
-    case WeededAst.Pattern.Cst(cst, loc) => NamedAst.Pattern.Cst(cst, loc)
+    case WeededAst.Pattern.Cst(cst, loc) => NamedAst.Pattern.Cst(cst, loc).toSuccess
 
     case WeededAst.Pattern.Tag(qname, pat, loc) =>
-      NamedAst.Pattern.Tag(qname, visitPattern(pat), loc)
+      mapN(visitPattern(pat)) {
+        case p => NamedAst.Pattern.Tag(qname, p, loc)
+      }
 
-    case WeededAst.Pattern.Tuple(elms, loc) => NamedAst.Pattern.Tuple(elms map visitPattern, loc)
+    case WeededAst.Pattern.Tuple(elms, loc) =>
+      mapN(traverse(elms)(visitPattern)) {
+        case es => NamedAst.Pattern.Tuple(es, loc)
+      }
 
+    case WeededAst.Pattern.Record(pats, pat, loc) =>
+      val psVal = traverse(pats) {
+        case WeededAst.Pattern.Record.RecordFieldPattern(field, tpe, pat1, loc1) =>
+          mapN(traverseOpt(tpe)(visitType), traverseOpt(pat1)(visitPattern)) {
+            case (t, p) => NamedAst.Pattern.Record.RecordFieldPattern(field, t, p, loc1)
```

Comment:
I think this is wrong. If `p` is `None` we should create a new pattern / symbol in the environment s.t. patterns like `{ field : Type }` or simply `{ field }` allow the following expression to use `field` as a variable, i.e.,
```scala
match { x = 1 } {
    case { x : Int32 } => println(x)
}
```

Reply:
Maybe the resolver also needs some rework to make this work